app/controllers/admin/students_controller.rb

```diff
@@ -22,13 +26,28 @@ def deactivate
   end
 
   def update
-    @cohort = Cohort.find(params[:cohort_id])
-    @student.update(cohort: @cohort)
-    render nothing: true if @student.save
+    if @student.update(student_params)
+      render nothing: true if request.xml_http_request?
+      redirect_to [:edit, :admin, @student], notice: 'Updated!' unless request.xml_http_request?
+    else
+      render :edit
```

Comment:
This will likely not work since you dont have `@cohorts` set (consider my `prep_form` private method approach?)